.github/workflows/main.yaml

```diff
@@ -85,21 +85,41 @@ jobs:
           for container in $(docker ps --format "table {{.Names}}" | grep -v "NAMES"); do
               echo "restart count for ${container} is $(docker inspect ${container} --format '{{json .RestartCount}}')"
           done
+      - name: Archive code coverage results
+        uses: actions/upload-artifact@v3
+        with:
+          name: code-coverage-result
+          path: |
+            ./go_coverage.txt
+            ./lcov_output.info
+            *.info
+            *.out
+            codecov.yml
+            internal/**
+            cmd/**
+            configs/**
+            pkg/**
+            go.mod
+            go.sum
+  codecov:
+    name: Upload Code Coverage
+    needs: UT
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - name:  Download code coverage results
+        uses: actions/download-artifact@v3.0.1
+        with:
+          name: code-coverage-result
+      - name: Display structure of code coverage results
+        run: ls -lah
       - name: Upload coverage to Codecov
         if: "github.repository == 'milvus-io/milvus'"
         uses: codecov/codecov-action@v3.1.1
         id: upload_cov
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
           files: ./go_coverage.txt,./lcov_output.info
-          name: ubuntu-${{ matrix.ubuntu }}-unittests
+          name: ubuntu-20.04-unittests
```

Comment:
do we need to hard code this os version here? does using `${{ matrix.ubuntu }}` cause any problem?

Reply:
the {{ matrix.ubuntu }} variable can only be used in the job UT, not able to be used in upload code coverage.